### PR TITLE
8283620: System.out does not use the encoding/charset specified in the Javadoc

### PR DESCRIPTION
**Pull request by naotoj**
Promoting the internal system properties for `System.out` and `System.err` so that users can override the encoding used for those streams to `UTF-8`, aligning to the `Charset.defaultCharset()`. A CSR has also been drafted.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8283620](https://bugs.openjdk.java.net/browse/JDK-8283620): System.out does not use the encoding/charset specified in the Javadoc
 * [JDK-8284778](https://bugs.openjdk.java.net/browse/JDK-8284778): System.out does not use the encoding/charset specified in the Javadoc (**CSR**)

### Reviewers
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**) ⚠️ Review applies to b5bcb87055523eae1e1df340d943f950ea767629

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8270/head:pull/8270` \
`$ git checkout pull/8270`

Update a local copy of the PR: \
`$ git checkout pull/8270` \
`$ git pull https://git.openjdk.java.net/jdk pull/8270/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8270`

View PR using the GUI difftool: \
`$ git pr show -t 8270`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8270.diff">https://git.openjdk.java.net/jdk/pull/8270.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/lang/System.java [hunk 1]
```diff
@@ -768,6 +768,16 @@ public static native void arraycopy(Object src,  int  srcPos,
...
+     * <tr><th scope="row">{@systemProperty stdout.encoding}</th>
+     *     <td>Character encoding name for {@link System#out System.out}.
+     *     The property may be set on the command line to the value
+     *     {@code UTF-8}. Setting the property to a value other than {@code UTF-8}
+     *     leads to unspecified behavior.
+     * <tr><th scope="row">{@systemProperty stderr.encoding}</th>
+     *     <td>Character encoding name for {@link System#err System.err}.
+     *     The property may be set on the command line to the value
+     *     {@code UTF-8}. Setting the property to a value other than {@code UTF-8}
+     *     leads to unspecified behavior.
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/lang/System.java:780]: I think the proposal to introduce two standard properties is good and is consistent with the recently introduced native.encoding. I'm not 100% sure that the sentence "The property may be set on the command line ..." is appropriate for the spec of standard properties. We got away with that for file.encoding in implNote but that isn't spec. I think we may have to replace this with something that says that the Java runtime can be started with the system property set to "UTF-8", starting it with the property set to another value clears to undefined behavior.

Comment (naotoj) [reply src/java.base/share/classes/java/lang/System.java]: Thanks, Alan. Modified them as suggested.

File: src/java.base/share/classes/java/lang/System.java [hunk 2]
```diff
@@ -768,6 +768,14 @@ public static native void arraycopy(Object src,  int  srcPos,
      * <tr><th scope="row">{@systemProperty native.encoding}</th>
      *     <td>Character encoding name derived from the host environment and/or
      *     the user's settings. Setting this system property has no effect.</td></tr>
+     * <tr><th scope="row">{@systemProperty stdout.encoding}</th>
+     *     <td>Character encoding name for {@link System#out System.out}.
+     *     The Java runtime can be started with the system property set to {@code UTF-8},
+     *     starting it with the property set to another value clears to undefined behavior.
```
Comment (dholmes-ora) [inline src/java.base/share/classes/java/lang/System.java:774]: _leads_ to undefined behavior

File: src/java.base/share/classes/java/lang/System.java (hunk 1 above)
Comment (dholmes-ora) [reply src/java.base/share/classes/java/lang/System.java]: I think Alan has a typo: "clears" -> "leads"

Comment (AlanBateman) [reply src/java.base/share/classes/java/lang/System.java]: > I think Alan has a typo: "clears" -> "leads"

Apologies, there was a typo in my comment on the PR. I didn't mean for the word "clears" to go into the spec.

**Review by AlanBateman — APPROVED** [review]
Overall I think this looks good, assuming the typo in the spec ("clears" -> "leads") is fixed.

**Pull request closed without merging**